xray/segment_test.go

```diff
@@ -325,3 +326,156 @@ func BenchmarkIdGeneration_noOpFalse(b *testing.B) {
 	}
 	os.Unsetenv("AWS_XRAY_NOOP_ID")
 }
+
+func TestSegment_TraceHeaderID(t *testing.T) {
+	type args struct {
+		traceHeader *header.Header
+	}
+	tests := []struct {
+		name    string
+		init    func(t *testing.T) *Segment
+		inspect func(r *Segment, t *testing.T) //inspects receiver after test run
+
+		args func(t *testing.T) args
+
+		want1 string
+	}{
```

Comment:
Also, we can move the test to `handler_test.go`. This is inline with above comments.